apps/managedidentity/managedidentity_test.go

```diff
@@ -48,95 +80,129 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
-		{code: http.StatusNotImplemented,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusServiceUnavailable,
-			err:           "",
-			desc:          "",
-			correlationid: "121212"},
-		{code: http.StatusBadRequest,
-			err:           "invalid_request",
-			desc:          "Identity not found",
-			correlationid: "121212",
-		},
+func setEnvVars(t *testing.T, source Source) {
+	switch source {
+	case AzureArc:
+		t.Setenv(IdentityEndpointEnvVar, "http://localhost:40342/metadata/identity/oauth2/token")
+		t.Setenv(ArcIMDSEnvVar, "http://localhost:40342 value")
+	case AppService:
+		t.Setenv(IdentityEndpointEnvVar, "identityEndpointEnvVar value")
+		t.Setenv(IdentityHeaderEnvVar, "identityHeaderEnvVar value")
+	case CloudShell:
+		t.Setenv(MsiEndpointEnvVar, "msiEndpointEnvVar value")
+	case ServiceFabric:
+		t.Setenv(IdentityEndpointEnvVar, "identityEndpointEnvVar value")
+		t.Setenv(IdentityHeaderEnvVar, "identityHeaderEnvVar value")
+		t.Setenv(IdentityServerThumbprintEnvVar, "identityServerThumbprintEnvVar value")
+	}
+}
+
+func unsetEnvVars() {
+	os.Unsetenv(IdentityEndpointEnvVar)
+	os.Unsetenv(IdentityHeaderEnvVar)
+	os.Unsetenv(IdentityServerThumbprintEnvVar)
+	os.Unsetenv(ArcIMDSEnvVar)
+	os.Unsetenv(MsiEndpointEnvVar)
+}
+
+func Test_Get_Source(t *testing.T) {
+	// todo update as required
+	testCases := []sourceTestData{
+		{name: "testAzureArcSystemAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: SystemAssigned()},
+		{name: "testAzureArcUserClientAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: UserAssignedClientID("clientId")},
+		{name: "testAzureArcUserResourceAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: UserAssignedResourceID("resourceId")},
+		{name: "testAzureArcUserObjectAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImds", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsEmptyEndpoint", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsLinux", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsEmptyEndpointLinux", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
 	}
 
 	for _, testCase := range testCases {
-		t.Run(http.StatusText(testCase.code), func(t *testing.T) {
-			fakeErrorClient := mock.Client{}
-			responseBody, err := makeResponseWithErrorData(testCase.err, testCase.desc)
-			if err != nil {
-				t.Fatalf("error while forming json response : %s", err.Error())
+		t.Run(string(testCase.source), func(t *testing.T) {
+			unsetEnvVars()
+			setEnvVars(t, testCase.source)
```

Comment:
[t.Setenv()](https://pkg.go.dev/testing#T.Setenv)

Reply:
Using the setEnvVars(t, testCase.source) instead as it sets based on the source, instead of having to set for each source twice